grails-app/controllers/com/careerfly/organization/interview/InterviewController.groovy

```diff
@@ -0,0 +1,195 @@
+import com.careerfly.user.User
+
+class InterviewController {
+
+    def index() { }
+
+    def jobRequirement() {
+
+        println params
+        session.company = params
+    }
+
+    def save() {
+
+        println params
+
+        Interview interviewInstance = new Interview()
+
+        Set workExperience = params.workExperience.split(";")
+        Set tech = params.technologies.split(";")
+        Set tools = params.tools.split(";")
+        Set skills = params.skills.split(";")
+
+        City cityInstance = new City([name: session.company.address])
+        cityInstance.save()
+
+        Address addressInstance = new Address([city: cityInstance])
+        addressInstance.save()
+
+        Company companyInstance = new Company([name: session.company.name, address: addressInstance])
+        companyInstance.save()
+
+        tech.each {
+            Tag techInstance = Tag.findByName(it)
+            if (!techInstance) {
+                techInstance = new Tag(name: it)
+                techInstance.save()
+            }
+            println techInstance
+            println techInstance.errors
+            interviewInstance.technologies.add(techInstance)
+        }
+
+        tools.each {
+            Tag toolInstance = Tag.findByName(it)
+            if (!toolInstance) {
+                toolInstance = new Tag(name: it)
+                toolInstance.save()
+            }
+            interviewInstance.tools.add(toolInstance)
+        }
+
+        skills.each {
+            Tag skillInstance = Tag.findByName(it)
+            if (!skillInstance) {
+                skillInstance = new Tag(name: it)
+                skillInstance.save()
+            }
+            interviewInstance.skills.add(skillInstance)
+        }
+
+        Map getDetails = [company: companyInstance, candidate: session.loggedInUser, jobPosition: params.jobPosition,
+                          qualification: params.qualification, result:params.result, workExperience: workExperience,
+                          technologies: interviewInstance.technologies, tools: interviewInstance.tools,
+                          skills: interviewInstance.skills, rounds: params.rounds]
+
+        println getDetails
+
+        interviewInstance = new Interview(getDetails)
+        interviewInstance.save(flush: true)
+        println interviewInstance
+
+        redirect(action: 'show', id: interviewInstance.id)
+    }
+
+    def show() {
+
+        Interview interviewInstance = Interview.get(params.id)
+        [interviewInstance: interviewInstance]
+    }
+
+    def edit() {
+
+        Interview interviewInstance = Interview.get(params.id)
+
+        if (interviewInstance.candidate.id != session.loggedInUser) {
+            flash.message = "You are not allowed to edit this interview"
+            redirect(action: "index")
+            return
+        }
+
+        [interviewInstance: interviewInstance]
+    }
+
+    def update() {
+        
+        println params
```

Comment:
Two many print statements. Yes, print statement helps for logging but we can't have these much of print logs.